.github/workflows/clean-up-docker.yaml

```diff
@@ -0,0 +1,38 @@
+name: Clean up Docker Images on PR closure
+
+on:
+  workflow_call:
+
+# Permissions needed
+# permissions:
+#   packages: write
+
+jobs:
+  purge-image-storer:
+    name: Delete image from ghcr.io
+    runs-on: ubuntu-latest
+    steps:
+      - name: Delete image
+        uses: bots-house/ghcr-delete-image-action@v1.1.0
+        with:
+          owner: ${{ github.repository_owner }}
+          name: ${{ github.event.repository.name }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          tag: pr-${{ github.event.pull_request.number }}
+  clean-untagged-api:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to GitHub Container Registry with PAT_TOKEN
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Delete all containers from repository without tags
+        uses: quartx-analytics/ghcr-cleaner@v1
+        with:
+          owner-type: org # or user
+          token: ${{ secrets.PACKAGE_DELETE_PAT }}
+          repository-owner: ${{ github.repository_owner }}
+          package-name: ${{ github.event.repository.name }}
```

Comment:
let's make name as input parameter with default value to `${{ github.event.repository.name }}` so that the workflow can be used also for multi image repositories.